caffe2/operators/generate_proposals_op.cu

```diff
@@ -80,37 +95,38 @@ __global__ void GeneratePreNMSUprightBoxesKernel(
     dw = fmin(dw, bbox_xform_clip);
     dh = fmin(dh, bbox_xform_clip);
 
+    const float coord_offset(match_detection_output ? 0.0f : 1.0f);
+
     // Applying the deltas
-    float width = x2 - x1 + 1.0f;
+    float width = x2 - x1 + coord_offset;
     const float ctr_x = x1 + 0.5f * width;
     const float pred_ctr_x = ctr_x + width * dx; // TODO fuse madd
     const float pred_w = width * expf(dw);
     x1 = pred_ctr_x - 0.5f * pred_w;
-    x2 = pred_ctr_x + 0.5f * pred_w - 1.0f;
+    x2 = pred_ctr_x + 0.5f * pred_w - coord_offset;
 
-    float height = y2 - y1 + 1.0f;
+    float height = y2 - y1 + coord_offset;
     const float ctr_y = y1 + 0.5f * height;
     const float pred_ctr_y = ctr_y + height * dy;
     const float pred_h = height * expf(dh);
     y1 = pred_ctr_y - 0.5f * pred_h;
-    y2 = pred_ctr_y + 0.5f * pred_h - 1.0f;
+    y2 = pred_ctr_y + 0.5f * pred_h - coord_offset;
 
     // Clipping box to image
-    const float img_height = d_img_info_vec[3 * image_index + 0];
-    const float img_width = d_img_info_vec[3 * image_index + 1];
     const float min_size_scaled =
         min_size * d_img_info_vec[3 * image_index + 2];
-    x1 = fmax(fmin(x1, img_width - 1.0f), 0.0f);
-    y1 = fmax(fmin(y1, img_height - 1.0f), 0.0f);
-    x2 = fmax(fmin(x2, img_width - 1.0f), 0.0f);
-    y2 = fmax(fmin(y2, img_height - 1.0f), 0.0f);
+    x1 = fmax(fmin(x1, img_width - im_info_offset), 0.0f);
```

Comment:
for float32 values, use fminf and fmaxf

Reply:
Fixed here and otherwhere in the file generate_proposals_op.cu.